### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'initIrisRtc')

### ISSUE THREAD

**Issue by mdaniyal9** [AgoraIO-Extensions/Agora-Flutter-SDK#1434]
**Describe the bug**
When i try to join channel from agora flutter web SDK it crashes. Can anyone tell me what i am doing wrong?

**To Reproduce**
1. I have followed the steps in readme of agora flutter sdk repo
2. I have also added <script src="iris-web-rtc_n423_w4182_0.3.0.js"></script> in index.html
3. TypeError: Cannot read properties of undefined (reading 'initIrisRtc')
packages/agora_rtc_engine/src/impl/platform/web/js_iris_api_engine_binding_delegate.dart 20:5 createApiEngine
packages/iris_method_channel/src/platform/web/iris_method_channel_internal_web.dart 107:9 initilize
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54 runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 127:5 _async
packages/iris_method_channel/src/platform/web/iris_method_channel_internal_web.dart 103:41 initilize
packages/iris_method_channel/src/iris_method_channel.dart 48:42 initilize
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54 runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 127:5 _async
packages/iris_method_channel/src/iris_method_channel.dart 40:41 initilize
packages/agora_rtc_engine/src/impl/agora_rtc_engine_impl.dart 388:29 initialize$
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54 runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 127:5 _async
packages/agora_rtc_engine/src/impl/agora_rtc_engine_impl.dart 364:26 initialize
packages/go_router_example/pages/test-agora-call-new.dart 141:23 setupVideoSDKEngine

**Expected behavior**
As per documentation i should be able to join a channel

**Desktop (please complete the following information):**
 - OS: macOS (Sonoma 14.0)
 - Browser chrome(Version 119.0.6045.123 (Official Build) (arm64))

**Additional context**
I am running the app on chrome and using mac M1

**Comment by littleGnAl** [comment]
Can you share the index.html file?

**Comment by Adrianzctpa** [comment]
Literally the same problem as mine!
Here is my index.html as OP didn't provide his:
````
<!DOCTYPE html>
<html>
<head>
  <!--
    If you are serving your web app in a path other than the root, change the
    href value below to reflect the base path you are serving from.

    The path provided below has to start and end with a slash "/" in order for
    it to work correctly.

    Fore more details:
    * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base
  -->
  <base href="/">

  <meta charset="UTF-8">
  <meta content="IE=Edge" http-equiv="X-UA-Compatible">

  <!-- iOS meta tags & icons -->
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">

  <meta name="robots" content="noindex" />

  <!-- Favicon -->
  <link rel="icon" type="image/png" href="favicon.png" />

  <!-- Open Graph & SEO tags -->
  <meta property="og:description" content="Built with FlutterFlow." />
  <meta property="og:image" content="https://storage.googleapis.com/flutterflow-prod-hosting/og_splash_gradient.png" /> 

  <meta name="twitter:description" content="Built with FlutterFlow." />
  <meta name="twitter:image" content="https://storage.googleapis.com/flutterflow-prod-hosting/og_splash_gradient.png" />
  <meta name="twitter:card" content="summary_large_image" />

  <!-- Status Bar color in Safari browser (iOS) and PWA -->
  <meta name="theme-color" media="(prefers-color-scheme: light)" content="#7265e3">
  <meta name="theme-color" media="(prefers-color-scheme: dark)"  content="#1d2428">

  <link rel="manifest" href="manifest.json">
  <script>
    // The value below is injected by flutter build, do not touch.
    var serviceWorkerVersion = null;
  </script>
  <!-- This script adds the flutter initialization JS code -->
  <script src="flutter.js" defer></script>
</head>
<body>
  <script>
    
    window.addEventListener('load', function (ev) {
      // Download main.dart.js
      _flutter.loader.loadEntrypoint({
        entrypointUrl: 'main.dart.js',
        serviceWorker: {
          serviceWorkerUrl: 'flutter_service_worker.js?v=',
          serviceWorkerVersion: serviceWorkerVersion,
        },
        onEntrypointLoaded: async function(engineInitializer) {
          // Initialize the Flutter engine
          let appRunner = await engineInitializer.initializeEngine();
          // Run the app
          await appRunner.runApp();
        }
      });
    });
  </script>
  
  
  <script src="https://js.stripe.com/v3/"></script>
  <script src="iris-web-rtc_n423_w4182_0.3.0.js" type="application/javascript"></script>
  
</body>
</html>
````

**Comment by leilaabdel** [comment]
I am also getting the same error. This is my `index.html` file: 

```
<!DOCTYPE html>
<html>
<head>
  <!--
    If you are serving your web app in a path other than the root, change the
    href value below to reflect the base path you are serving from.

    The path provided below has to start and end with a slash "/" in order for
    it to work correctly.

    For more details:
    * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base

    This is a placeholder for base href that will be replaced by the value of
    the `--base-href` argument provided to `flutter build`.
  -->
  <base href="$FLUTTER_BASE_HREF">

  <meta charset="UTF-8">
  <meta content="IE=Edge" http-equiv="X-UA-Compatible">
  <meta name="description" content="A new Flutter project.">

  <!-- iOS meta tags & icons -->
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  <meta name="apple-mobile-web-app-title" content="quranic_web">
  <link rel="apple-touch-icon" href="icons/Icon-192.png">

  <!-- Favicon -->
  <link rel="icon" type="image/png" href="favicon.png"/>

  <title>quranic_web</title>
  <link rel="manifest" href="manifest.json">

  <script>
    // The value below is injected by flutter build, do not touch.
    var serviceWorkerVersion = null;
  </script>
  <!-- This script adds the flutter initialization JS code -->
  <script src="flutter.js" defer></script>
</head>
<body>
  <script>
    window.addEventListener('load', function(ev) {
      // Download main.dart.js
      _flutter.loader.loadEntrypoint({
        serviceWorker: {
          serviceWorkerVersion: serviceWorkerVersion,
        },
        onEntrypointLoaded: function(engineInitializer) {
          engineInitializer.initializeEngine().then(function(appRunner) {
            appRunner.runApp();
          });
        }
      });
    });
  </script>

  <script src="https://download.agora.io/staging/iris-web-rtc_n423_w4182_0.3.0.js"></script>

</body>
</html>

```

The error message is as follows: `Error during initialize: TypeError: Cannot read properties of undefined (reading 'initIrisRtc')`

**Comment by littleGnAl** [comment]
Hello @Adrianzctpa @leilaabdel, I tried your html config in our example, and it works fine.
Is possible to share a reproducible demo? And can you share the `flutter doctor -v`?

**Comment by mdaniyal9** [comment]
> Hello @Adrianzctpa @leilaabdel, I tried your html config in our example, and it works fine. Is possible to share a reproducible demo? And can you share the `flutter doctor -v`?

Hi littleGnAl i cloned your SDK repo and gave path to package just like you have done in repo and my example started working 
then i removed the Path of package from your repo and gave it the latest version 6.2.4, after that your code also gave the same error as mine was giving before.
So i think issue is in the SDK which is published with version 6.2.4
**Working**
  agora_rtc_engine:
    path: ../../CHI/Examples/Agora-Flutter-SDK  
    
**Not Working**
  agora_rtc_engine: ^6.2.4

**Comment by littleGnAl** [comment]
The code has landed in the `main` branch but has not been published to the `pub.dev` yet, so 6.2.4 not included this feature yet, you should be dependent on the main branch as a git package at this time.

Add the main branch as git package:
```
agora_rtc_engine:
  git:
    url: https://github.com/AgoraIO-Extensions/Agora-Flutter-SDK.git
    ref: main
```

Then follow the integration guide https://github.com/AgoraIO-Extensions/Agora-Flutter-SDK?tab=readme-ov-file#web-alpha

**Comment by mdaniyal9** [comment]
> git:
>     url: https://github.com/AgoraIO-Extensions/Agora-Flutter-SDK.git
>     ref: main

I have added the above in my pubspec but it stucks on Resolving dependencies...
if i revert to version then flutter pub get works fine but with git url it's not.

i have also tried another package with git it works.

**Comment by mdaniyal9** [comment]
> The code has landed in the `main` branch but has not been published to the `pub.dev` yet, so 6.2.4 not included this feature yet, you should be dependent on the main branch as a git package at this time.
> 
> Add the main branch as git package:
> 
> ```
> agora_rtc_engine:
>   git:
>     url: https://github.com/AgoraIO-Extensions/Agora-Flutter-SDK.git
>     ref: main
> ```
> 
> Then follow the integration guide https://github.com/AgoraIO-Extensions/Agora-Flutter-SDK?tab=readme-ov-file#web-alpha

Now it's Working, thanks

**Comment by developersTeem** [comment]
@littleGnAl  I have checked this solution and still I am getting the same error please find the below pubspec.yaml and pubspec.lock and let me know if is there anything wrong

**pubspec.yaml**
  agora_rtc_engine:
    git:
      url: https://github.com/AgoraIO-Extensions/Agora-Flutter-SDK.git
      ref: main

**pubspec.lock**
agora_rtc_engine:
    dependency: "direct main"
    description:
      path: "."
      ref: main
      resolved-ref: bee47d50c6a27446522b3df73590215d640ecf6a
      url: "https://github.com/AgoraIO-Extensions/Agora-Flutter-SDK.git"
    source: git
    version: "6.2.4"
